Metalama.Framework/Code/SyntaxBuilders/IStatementList.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+namespace Metalama.Framework.Code.SyntaxBuilders;
+
+/// <summary>
+/// Represents a list of statements. This list cannot be enumerated because it is lazily evaluated. To create an
+/// <see cref="IStatementList"/>, use <see cref="StatementFactory.List(Metalama.Framework.Code.SyntaxBuilders.IStatement[])"/>,
+///  <see cref="StatementFactory.UnwrapBlock"/>, or <see cref="StatementExtensions.AsList(Metalama.Framework.Code.SyntaxBuilders.IStatement)"/>
+/// has been added for convenience.
```

Comment:
I think the grammar here doesn't make sense.

Either:

```suggestion
/// Represents a list of statements. This list cannot be enumerated because it is lazily evaluated. To create an
/// <see cref="IStatementList"/>, use <see cref="StatementFactory.List(Metalama.Framework.Code.SyntaxBuilders.IStatement[])"/> or
///  <see cref="StatementFactory.UnwrapBlock"/>. <see cref="StatementExtensions.AsList(Metalama.Framework.Code.SyntaxBuilders.IStatement)"/>
/// has been added for convenience.
```

Or:

```suggestion
/// Represents a list of statements. This list cannot be enumerated because it is lazily evaluated. To create an
/// <see cref="IStatementList"/>, use <see cref="StatementFactory.List(Metalama.Framework.Code.SyntaxBuilders.IStatement[])"/>,
///  <see cref="StatementFactory.UnwrapBlock"/>, or <see cref="StatementExtensions.AsList(Metalama.Framework.Code.SyntaxBuilders.IStatement)"/>.
```